go-controller/pkg/ovn/service.go

```diff
@@ -169,6 +169,21 @@ func (ovn *Controller) createService(service *kapi.Service) error {
 		return nil
 	}
 
+	// We can end up in a situation where the endpoint creation is triggered before the service creation,
+	// we should not be creating the reject ACLs if this endpoint exists, because that would result in an unreachable service
+	// eventough the endpoint exists.
+	// NOTE: we can also end up in a situation where a service matching no pods is created. Such a service still has an endpoint, but with no subsets.
+	// make sure to treat that service as an ACL reject.
+	ep, err := ovn.watchFactory.GetEndpoint(service.Namespace, service.Name)
+	if err == nil {
+		if len(ep.Subsets) > 0 {
+			klog.V(5).Infof("service has endpoint, will create loadbalancer VIPs for service: %s", service.Name)
+		} else {
+			klog.V(5).Infof("service has empty endpoint, will create ACL reject for service: %s", service.Name)
```

Comment:
this isn't technically true, because we don't know if the service qualifies for reject, but I'm fine with it.

Reply:
I can remove the "will create ACL reject..." part